user-guide/Cloud_Platform/Catalog/Register_Catalog_Item.md

```diff
@@ -0,0 +1,139 @@
+
+The call accepts a binary file as input. This file should be a zip file containing a ["manifest.yml"](xref:Register_Catalog_Item#manifest-file) and the item you want to upload.
+
+#### Manifest file
+
+This file will contain all necessary information to register a Catalog item or version. This manifest file should be a valid yml file and will contain "required" and "optional" attributes to add extra information to the Catalog item. Note that limitations may apply to certain attributes based on length or formatting.
+
+```yml
+# [Required]
+# Possible values: 
+#   - AutomationScript: If the catalog item is an Automation Script that can be deployed on a DataMiner System.
+#   - AppPackage: If the catalog item is a Solution Package that can be deployed on a DataMiner System.
+#   - ProtocolPackage: If the catalog item is a Protocol that can be deployed on a DataMiner System.
+#   - Visio: If the catalog item is a Visio that can be deployed on a DataMiner System.
+type: 'AutomationScript'
+
+# [Required] 
+# The id of the catalog item.
+# All versions for the same id are shown together in the catalog.
+# This id can not be changed. 
+# If the id is not filled in, the API will return one and consider this as the registration of a new item.
+# If the id is filled in, a new version will be added to the catalog item with the given version number.
+#   Must be a valid GUID.
+id: '<fill in guid here or leave empty>'
+
+# [Required] 
+# The human-friendly name of the catalog item. 
+# Can be changed at any time.
+#   Max length: 100 characters.
+#   Can not contain newlines.
+#   Can not contain leading or trailing whitespace characters.
+title: 'add-new-switch'
+
+# [Required] 
+# The version in format X.X.X
+# Once a version has been published it can no longer be changed.
+# The version should be incremented to do any changes.
+# If you want to create a package that is not final, you can provide a label at the end: X.X.X-label
+# If a catalog item has a label in the version, then the version is considered to be a pre-release version, and not an official one.
+#   Valid versions range from 0.0.0 to 2147483647.2147483647.2147483647
+#   The label can contain characters a-z0-9 with a max length of 30 characters.
+version: '1.0.0'
+
+# [Optional]
+# General information about the catalog item.
+#   Max length: 100.000 characters
+description: |
+  This script allows you to easily add a new leaf switch element.
+
+# [Optional]
+# A short description for this specific version.
+# This message is shown along side each version in the version overviews in the catalog. 
+#   Max length: 500 characters.
+version-message: 'Initial version. Does not contain anything useful yet.'
+
+# [Optional]
+# A valid URL that points to the home page.
+# Usually, this home page is also the location of the source code.
```

Comment:
Can we somehow automatically use the 'URL' when we 'tag' a catalog item to be released and use that link? It feels counter-productive to specify the link manually.

Reply:
We could detect this in the github action but note that maybe a user will want to point to external documentation so this is why we allow this.